packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -845,19 +853,17 @@ - (void)stopVideoRecordingWithCompletion:(void (^)(NSString *_Nullable,
   if (_isRecording) {
     _isRecording = NO;
 
-    if (_videoWriter.status != AVAssetWriterStatusUnknown) {
```

Comment:
why is this check removed?

Reply:
This is where it was not sending completion (in false case) so a dart future never resolved. But now `startWriting` is called at the same time as `_isRecording` is set to `YES` so `finishWritingWithCompletionHandler` in `stopVideoRecordingWithCompletion` can be called only after was called `startWriting` which sets `_videoWriter.status` to `AVAssetWriterStatusWriting` so that check redundant (and again it would just indicate false possibility of not calling completion).

Reply:
Actually maybe it should check whether status is `AVAssetWriterStatusWriting` and complete with error otherwise because it can be for example `AVAssetWriterStatusFailed` here (I suppose `finishWritingWithCompletionHandler ` would just crash in that case). But this situation is nothing new introduced with this PR as before there was just check `status != unknown` which would pass `failed` to `finishWritingWithCompletionHandler`.

I also noticed check for `AVAssetWriterStatusFailed` in `didOutputSampleBuffer` where is called `[self reportErrorMessage]` but when I tried to force call `reportErrorMessage` I did not see this error report anywhere.

Reply:
Regarding previous comment, seems `finishWritingWithCompletionHandler` is ok with `AVAssetWriterStatusFailed` it will just pass it to handler so there is no need to check whether is status not `failed` before calling it, it only has problem with `AVAssetWriterStatusCancelled` but there is no call to `cancelWriting`, `AVAssetWriterStatusCompleted` and `AVAssetWriterStatusUnknown` but these two also cannot happen.